src/alg_utils.jl

```diff
@@ -1,8 +1,8 @@
-isautodifferentiable(alg::OrdinaryDiffEqAlgorithm) = true
+isautodifferentiable(alg::Union{OrdinaryDiffEqAlgorithm,DAEAlgorithm}) = true
```

Comment:
Would it make sense to rename `OrdinaryDiffEqAlgorithm` to `ODEAlgorithm` for consistency? And introduce an alias for `Union{OrdinaryDiffEqAlgorithm,DAEAlgorithm}`?

Reply:
Agreed. What should we call the Union?

Reply:
If `ODEAlgorithm` is introduced, maybe just `OrdinaryDiffEqAlgorithm` since `ODEAlgorithm` and `DAEAlgorithm` are defined by OrdinaryDiffEq?